src/backend/distributed/commands/index_pg_source.c

```diff
@@ -0,0 +1,195 @@
+/*-------------------------------------------------------------------------
+ *
+ * index_pg_source.c
+ *    Helper functions copy & pasted from PostgreSQL source code.
+ *    All the functions in this file is copied from
+ *    postgres/src/backend/commands/indexcmds.c
+ *
+ * Copyright (c) Citus Data, Inc.
```

Comment:
For proper attribution, instead of mentioning `Citus Data` copyright, I think we should attribute pg license for this file like we do in ruleutils files

https://github.com/citusdata/citus/blob/a58a4395abec1e38d83629b66d1b33766c040897/src/backend/distributed/deparser/ruleutils_13.c#L7-L12